kolibri/core/assets/src/api-resource.js

```diff
@@ -237,17 +249,16 @@ class Collection {
         } else {
           this.synced = false;
           this.resource.client({ path: this.url, params }).then((response) => {
-            // Reset current models to only include ones from this fetch.
-            this.models = [];
```

Comment:
looks like these lines below also get called in the case of malformed data - is that correct?

```javascript
            this.synced = true;
            this.models.forEach((model) => {
              model.synced = true; // eslint-disable-line no-param-reassign
            });
```